src/utp_stream.cpp

```diff
@@ -89,6 +88,29 @@ void utp_log(char const* fmt, ...)
 	va_end(l);
 }
 
+bool is_utp_stream_logging() {
+	return log_file_holder.utp_log_file != NULL;
+}
+
+void set_utp_stream_logging(bool enable) {
+	if (enable)
+	{
+		if (log_file_holder.utp_log_file == NULL)
+		{
+			log_file_holder.utp_log_file = fopen("utp.log", "w+");
+		}
+	}
+	else
+	{
+		if (log_file_holder.utp_log_file != NULL)
+		{
+			FILE* f = log_file_holder.utp_log_file;
+			log_file_holder.utp_log_file = NULL;
+			fclose(f);
+		}
+	}
+}
+
 #define UTP_LOG utp_log
 #if TORRENT_VERBOSE_UTP_LOG
```

Comment:
all uses of TORRENT_VERBOSE_UTP_LOG and TORRENT_UTP_LOG use "#if", and not "#ifdef". That will cause warnings the way it is now, because these won't be defined sometimes. Either make them always defined (and sometimes as 0) or change the tests of them to use "#ifdef"